File: Data Structures & Algorithms/frequency-of-the-most-frequent-element/submission-1.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        # 124 5
        # 224 4
        # 324 3
        # 424 2
        # 434 1
        # 444 1
# K = 5
        # 1 4 8 13 - 5
        # 2 4 8 13 - 4
        # 3 4 8 13 - 3
        # 4 4 8 13 - 2
        nums.sort()
        res = 0
        total = 0
        l, r = 0, 0

        while r < len(nums):
            total += nums[r]

            while nums[r] * (r -l + 1) > total + k:
                total -= nums[l]
                l += 1
            res = max(res, r - l + 1)
            r+=1

        return res

File: Data Structures & Algorithms/three-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {
    nums.sort((a, b) => a - b); // Sort the array to handle duplicates easily
    const target = 0;
    const results = [];

    for (let i = 0; i < nums.length - 2; i++) {
        // Skip duplicates for the first number
        if (i > 0 && nums[i] === nums[i - 1]) continue;

        let left = i + 1, right = nums.length - 1;

        while (left < right) {
            const sum = nums[i] + nums[left] + nums[right];
            if (sum === target) {
                results.push([nums[i], nums[left], nums[right]]);

                // Skip duplicates for the second number
                while (left < right && nums[left] === nums[left + 1]) left++;
                // Skip duplicates for the third number
                while (left < right && nums[right] === nums[right - 1]) right--;

                left++;
                right--;
            } else if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
    }

    return results;
    }
}
